marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/entities/entity-search-lib.sjs

```diff
@@ -535,11 +540,58 @@ function isHubEntityInstance(docUri) {
   return isHubEntityInstance;
 }
 
+function getRecordHistory(docUri) {
+  const history = [];
+  const provResults = xdmp.invokeFunction(function () {
+    const match = {
+      attributes: {
+        location: docUri
+      }
+    };
+    const out = {
+      dateTime: '?',
+      relations: {'associatedWith': '?', 'attributedTo': '?'}
+    };
+    const kvPattern = ps.opTriplePattern(match, out);
+    return op.fromTriples(kvPattern).result();
+  }, { 'database' : xdmp.database(datahub.config.JOBDATABASE)}).toArray();
+
+  if(provResults.length == 0) {
+    const metadata = xdmp.documentGetMetadata(docUri);
+    if(metadata) {
+      const currentObject = {};
+      currentObject.updatedTime = metadata.datahubCreatedOn ? metadata.datahubCreatedOn : undefined;
+      currentObject.flow = metadata.datahubCreatedInFlow ? metadata.datahubCreatedInFlow : undefined;
+      currentObject.step = metadata.datahubCreatedByStep ? metadata.datahubCreatedByStep : undefined;
+      currentObject.user = metadata.datahubCreatedBy ? metadata.datahubCreatedBy : undefined;
+      history.push(currentObject);
+    }
+  } else {
+    const flowNames = fn.collection("http://marklogic.com/data-hub/flow").toArray().map(flow => JSON.stringify(flow.toObject().name));
+    const stepNames = fn.collection(["http://marklogic.com/data-hub/steps", "http://marklogic.com/data-hub/step-definition"]).toArray().map(step => JSON.stringify(step.toObject().name));
```

Comment:
- Should we be ignoring "hub-core-artifacts" for both `flowNames` and `stepNames` here?
- We can use a Set for `flowNames` and `stepNames` instead of an array.
- No need to `JSON.stringify` here since `name` is already a string.